src/Resources/ResourceManager/Implementation/ResourceGroups/ExportAzureResourceGroupCmdlet.cs

```diff
@@ -93,63 +96,90 @@ public class ExportAzureResourceGroupCmdlet : ResourceManagerCmdletBaseWithApiVe
-
-                var template = JToken.FromObject(JObject.Parse(resultString)["template"]);
-
-                if (JObject.Parse(resultString)["error"] != null)
+                    var parameters = new Management.ResourceManager.Models.ExportTemplateRequest
+                    {
+                        Resources = this.GetResourcesFilter(resourceGroupId: resourceGroupId),
+                        Options = this.GetExportOptions(),
+                    };
+
+                    var exportedTemplate = ResourceManagerSdkClient.ExportResourceGroup(ResourceGroupName, parameters);
+
+                    var template = exportedTemplate.Template;
+                    contents = template.ToString();
+
+                    var error = exportedTemplate.Error;
+                }
+                else
                 {
-                    ExtendedErrorInfo error;
-                    if (JObject.Parse(resultString)["error"].TryConvertTo(out error))
+                    var parameters = new ExportTemplateParameters
+                    {
+                        Resources = this.GetResourcesFilter(resourceGroupId: resourceGroupId),
+                        Options = this.GetExportOptions(),
+                    };
+                    var apiVersion = this.ApiVersion ?? DefaultApiVersion;
+                    var operationResult = this.GetResourcesClient()
+                       .InvokeActionOnResource<JObject>(
+                           resourceId: resourceGroupId,
+                           action: Constants.ExportTemplate,
+                           parameters: parameters.ToJToken(),
+                           apiVersion: apiVersion,
+                           cancellationToken: this.CancellationToken.Value)
+                       .Result;
+
+                    var managementUri = this.GetResourcesClient()
+                        .GetResourceManagementRequestUri(
+                            resourceId: resourceGroupId,
+                            apiVersion: apiVersion,
+                            action: Constants.ExportTemplate);
+
+                    var activity = string.Format("POST {0}", managementUri.PathAndQuery);
+                    var resultString = this.GetLongRunningOperationTracker(activityName: activity,
+                        isResourceCreateOrUpdate: false)
+                        .WaitOnOperation(operationResult: operationResult);
+
+                    var template = JToken.FromObject(JObject.Parse(resultString)["template"]);
+                    contents = template.ToString();
+
+                    if (JObject.Parse(resultString)["error"] != null)
                     {
-                        WriteWarning(string.Format("{0} : {1}", error.Code, error.Message));
-                        foreach (var detail in error.Details)
+                        ExtendedErrorInfo error;
+                        if (JObject.Parse(resultString)["error"].TryConvertTo(out error))
```

Comment:
A little C# magic to make life easier :)
```suggestion
                        if (JObject.Parse(resultString)["error"].TryConvertTo(out ExtendedErrorInfo error))
```